source/ota.c

```diff
@@ -285,12 +285,16 @@ static OtaJobParseErr_t validateAndStartJob( OtaFileContext_t * pFileContext,
 /**
  * @brief Parse the OTA job document, validate and return the populated OTA context if valid.
  *
+ * @param[in] pJsonDoc Structure to store the details of keys and where to store them.
+ * @param[in] numJobParams Number of parameters to be extracted.
  * @param[in] pJson JSON job document.
  * @param[in] messageLength Length of the job document.
  * @param[in] pUpdateJob Represents if the job is accepted.
  * @return OtaFileContext_t* File context to store file information.
  */
-static OtaFileContext_t * parseJobDoc( const char * pJson,
+static OtaFileContext_t * parseJobDoc( const JsonDocParam_t * pJsonDoc,
```

Comment:
the name "pJsonDoc" is misleading compared to the description. This sounds like a pointer to a JSON document. But this is a pointer to a struct that has information about the expected structure of the JSON document. Maybe change it to something like "pJsonDocStructure"?

Reply:
Changed to `pJsonExpectedParams`